server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -160,8 +167,7 @@ static boolean isIdleThread(ThreadInfo threadInfo) {
             if (cpuTime == INVALID_TIMING) {
                 continue;
             }
-            //put to result when getThreadAllocatedBytes return -1
-            long allocatedBytes = SunThreadInfo.getThreadAllocatedBytes(threadIds[i]);
+            long allocatedBytes = type == ReportType.MEM ? sunThreadInfo.getThreadAllocatedBytes(threadIds[i]) : 0;
```

Comment:
Add an `SunThreadInfo sunThreadInfo` argument to getAllValidThreadInfos right after ThreadMxBean to pass either the real sunThreadInfo or the mocked one.